### ISSUE TITLE
Change event causes Vim cursor to reset

### ISSUE THREAD

**Issue by KarlClinckspoor** [Zettlr/Zettlr#905]
## Description

When editing in vim mode, the cursor would go from | to a block after some time. The file is also reloaded. The notes are in a Dropbox folder, but I told it to pause synchronization. Possibly related to #822? This happens with the following lines in the logs:

```
[10:24:52]	>>> IPC IN: file-save
[10:24:52]	Trying to run command through Application: file-save
[10:24:52]	<<< IPC OUT: tags-database
[10:24:52]	<<< IPC OUT: tags-database
[10:24:52]	<<< IPC OUT: mark-clean
[10:24:52]	<<< IPC OUT: file-replace
[10:24:52]	Chokidar has detected a change event for file 2020_06_09_10-18-10 test.md. Attempting to re-parse ...
[10:24:52]	<<< IPC OUT: tags-database
[10:24:52]	<<< IPC OUT: sync-files
[10:24:52]	The translation string was malformed: Open Logs!
[10:24:52]	<<< IPC OUT: config-update
[10:24:52]	<<< IPC OUT: sync-files
[10:24:52]	The translation string was malformed: Open Logs!
[10:24:52]	<<< IPC OUT: config-update
[10:24:52]	<<< IPC OUT: file-replace
[10:24:52]	<<< IPC OUT: replace-file-contents
[10:27:18]	<<< IPC OUT: export
```

## Reproducing
This is what I did to reproduce this:

1. I created a new file.
2. Set edit to vim mode.
3. I would enter insert mode with `i`, type anything, Alt-Tab to the logs window, wait a second for it to autosave, then it would show those messages.
4. Go back to the main zettlr window by alt-tabbing. The cursor would be a block, at the start of the file, and in insert mode. Pressing escape goes back to normal mode, and the character remains a block, and pressing `i` gets back to insert mode, with `|` as a cursor. 

I've included two screenshots, although they did not capture the cursor symbol.

## Expected behaviour
The type of character should be more tied to the mode it's at the moment, and the reloading is also important to fix.

**Screenshots**

![Right after Alt-tabbing (some time to take the screenshot though)](https://user-images.githubusercontent.com/30571394/84153149-e6c9d200-aa3b-11ea-8a44-4ca6009e73f2.PNG)

![After typing](https://user-images.githubusercontent.com/30571394/84153154-e7faff00-aa3b-11ea-8eed-51f0474e5a33.PNG)

## Platform
 - OS and version: Windows 8.1
 - Zettlr Version: 1.7.0-beta10
 - Screen Resolution: 1080p

**Comment by nathanlesage** [comment]
Does this still happen in the most recent beta?

**Comment by KarlClinckspoor** [comment]
I have tested it now with the newest beta, and didn't have any issues. I do get a lot of `[WATCHDOG] Ignore event: change:` events, which I think is what you were working on, right? However, I downgraded to previous versions (through git checkout <id of a previous release> and yarn install, should be ok, right?) and I wasn't able to replicate the bug, even though I could get it to trigger basically at will every minute a few days ago. I'll continue working in the next few days and report if I find anything.

**Comment by nathanlesage** [comment]
alright, thanks!

**Comment by KarlClinckspoor** [comment]
So, I managed to get this to happen by accident, while recording, but haven't been able to reproduce it. However, the following appeared in the logs when this bug happened.

```
>>> IPC IN: file-save
[10:00:28]	Trying to run command through Application: file-save
[10:00:28]	[WATCHDOG] Ignore event: change:C:\Users\user\Dropbox\Zettelkasten\2020_05_05_15-13-30 Viscosity correlations.md
[10:00:28]	<<< IPC OUT: tags-database
[10:00:28]	<<< IPC OUT: tags-database
[10:00:28]	<<< IPC OUT: mark-clean
[10:00:28]	<<< IPC OUT: file-replace
[10:00:28]	[WATCHDOG] Emitting event: change:C:\Users\user\Dropbox\Zettelkasten\2020_05_05_15-13-30 Viscosity correlations.md
isDir: false
ignoreDir: false
isFile: true
ignoreFile: false
isAttachment: false
[10:00:28]	[FSALFile::hasChangedOnDisk] File 2020_05_05_15-13-30 Viscosity correlations.md has modification time 1592917144170, and fs.lstat reports 1592917228210
[10:00:28]	Change event detected. FSALFile::hasChangedOnDisk reports: true with modtime 1592917144170
parent: -1853596831
dir: C:\Users\user\Dropbox\Zettelkasten
path: C:\Users\user\Dropbox\Zettelkasten\2020_05_05_15-13-30 Viscosity correlations.md
name: 2020_05_05_15-13-30 Viscosity correlations.md
hash: 2015271330
ext: .md
id: 2020_05_05_15-13-30
tags: ["rheology","polymer","viscosity","molar-mass","intrinsic-viscosity","property-correlations"]
type: file
wordCount: 510
charCount: 3368
target: undefined
modtime: 1592917144170
creationtime: 1588702412461
frontmatter: undefined
linefeed:
modified: false
[10:00:28]	Chokidar has detected a change event for file 2020_05_05_15-13-30 Viscosity correlations.md. Attempting to re-parse ...
[10:00:28]	<<< IPC OUT: tags-database
[10:00:28]	<<< IPC OUT: sync-files
[10:00:28]	The translation string was malformed: Open Logs!
[10:00:28]	<<< IPC OUT: config-update
[10:00:28]	<<< IPC OUT: sync-files
[10:00:28]	The translation string was malformed: Open Logs!
[10:00:28]	<<< IPC OUT: config-update
[10:00:28]	<<< IPC OUT: file-replace
[10:00:28]	<<< IPC OUT: replace-file-contents
```

[Here's the recording](https://www.youtube.com/watch?v=veBxZkrSfjM&feature=youtu.be). This happened entirely by accident, so I wasn't prepared. What I recall having happened was that I typed `ESC`, then `b`, then `cw` (which gets me into insert mode, as the cursor becomes `|`), then waited (I was actually reading something else to type), then apparently the file saved, that stuff in the log happened, I wrote `and`, was surprised that the cursor reverted to a block, pressed `ESC`, then stopped the recording.

I tried reproducing this, but the watchdog ignored all the events. So this seems very rare, and your fix did make these reloads much rarer. Notice also that the cursor did not jump to the beginning of the file, as it happened previously. I am using the latest beta (v1.7.0-beta.13). Sorry if I haven't answered before, I didn't use Zettlr much in the last few days.

**Comment by KarlClinckspoor** [comment]
Note, it happened again! Same log result. Maybe dragging the mouse over to another window made this happen?

[Here's my recording](https://youtu.be/jphJ9zEDCkg)

**Comment by nathanlesage** [comment]
> Notice also that the cursor did not jump to the beginning of the file, as it happened previously.

This was a different error because Zettlr didn't attempt to preserve the cursor position previously, but that got fixed as well.

As for the bug … I'm still at a loss. Currently I assume that on Windows systems, weird race conditions lead to chokidar quite literally "choking", which is why I added the additional setting. Apparently, times even beyond 2 seconds of delay may be necessary to circumvent this bug (that is: On Windows sometimes change events are fired even if the process saving the data is not yet done writing data; which means that the new feature introduced in the last beta is meant to resolve that. The amount of milliseconds actually tells Zettlr's watchdog to wait for at least that amount of time until a change event will be registered as such)

Did you make use of that setting already? (It's in the advanced tab of the preferences)
